source/agora/consensus/data/TransactionData.d

```diff
@@ -0,0 +1,297 @@
+/*******************************************************************************
+
+    Contains a type for storing data in a transaction
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.consensus.data.TransactionData;
+
+import agora.common.Hash;
+import agora.common.Serializer;
+
+import std.algorithm.comparison;
+import std.algorithm.iteration;
+import std.range;
+import std.utf;
+
+/// The structure of for storing data in a transaction
+public struct TransactionData
+{
+    /// The byte array of transaction data
+    public ubyte[] data;
+
+    /// The error for fromString(),
+    /// This will be checked in the validation of the transaction.
+    public string error;
```

Comment:
Errors should not be embedded in the structures that will be stored in the UTXO set or in the blockchain.

Reply:
I changed the function fromString to solve this problem.